File: DataSets/DataSet-Old/LeetCodeTasksDebug/Fixed/60_permutation-sequence/solutions/python3/solution.py
class Solution:
    def getPermutation(self, n: int, k: int) -> str:
        from math import factorial

        numbers = list(range(1, n + 1))
        permutation = ""

        k -= 1  # convert k from 1-based index to 0-based index for easier calculations

        for i in range(n, 0, -1):
            fact = factorial(i - 1)
            index = k // fact
            permutation += str(numbers[index])
            numbers.pop(index)
            k %= fact

        return permutation

File: DataSets/DataSet-Old/LeetCodeTasksDebug/Fixed/60_permutation-sequence/solutions/c/solution.c
#include <stdio.h>
#include <stdlib.h>

char* getPermutation(int n, int k) {
    char *result = (char *)malloc((n + 1) * sizeof(char));
    int factorial[n + 1];
    factorial[0] = 1;
    for (int i = 1; i <= n; i++) {
        factorial[i] = factorial[i - 1] * i;
    }

    k--; // Convert k from 1-based index to 0-based index for easier calculation.
    char available[10];
    for (int i = 0; i < n; i++) {
        available[i] = '1' + i; // Initialize the list of available characters.
    }

    for (int i = 0; i < n; i++) {
        int idx = k / factorial[n - 1 - i];
        result[i] = available[idx];
        // Remove used character by shifting the rest
        for (int j = idx; j < n - 1; j++) {
            available[j] = available[j + 1];
        }
        k %= factorial[n - 1 - i];
    }

    result[n] = '\0'; // Null-terminate the string.
    return result;
}
